contracts/zero-ex/tests/Multiplex.t.sol

```diff
@@ -0,0 +1,1453 @@
+        mintTo(address(tokenB), address(pool), balanceB);
+    }
+
+    function encodeFractionalFillAmount(uint256 frac) private pure returns (uint256) {
+        return HIGH_BIT + (frac * 1e16);
+    }
+
+    function describeLog(
+        bytes32[] memory topics,
+        bytes memory data,
+        bytes memory dataMask
+    ) private pure returns (LogDescription memory) {
+        assert(dataMask.length == 0 || dataMask.length == data.length);
+        return LogDescription(topics, data, dataMask);
+    }
+
+    function describeLog(bytes32[] memory topics, bytes memory data) private pure returns (LogDescription memory) {
+        return describeLog(topics, data, hex"");
+    }
+
+    function assertLogs(LogDescription[] memory expected) private {
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+
+        uint256 j = 0;
+        for (uint256 i = 0; i < entries.length && j < expected.length; i++) {
+            Vm.Log memory en = entries[i];
+            LogDescription memory ex = expected[j];
+            uint256 k;
+
+            if (en.topics.length != ex.topics.length || en.data.length != ex.data.length) continue;
+
+            for (k = 0; k < en.data.length; k++) {
+                if (ex.dataMask.length > 0) {
+                    en.data[k] = en.data[k] & ex.dataMask[k];
+                    ex.data[k] = ex.data[k] & ex.dataMask[k];
+                }
+
+                if (en.data[k] != ex.data[k]) {
+                    k = en.data.length + 1;
+                    break;
+                }
+            }
+
+            if (k != en.data.length) continue;
+
+            for (k = 0; k < en.topics.length; k++) {
+                if (en.topics[k] != ex.topics[k]) k = en.topics.length + 1;
+            }
+
+            if (k != en.topics.length) continue;
+
+            j++;
+        }
+
+        if (j != expected.length) {
+            fail("logs do not match expected");
+            for (uint256 i = 0; i < expected[j].topics.length; i++)
+                log_named_bytes32("expected topic", expected[j].topics[i]);
+            log_named_bytes("expected data", expected[j].data);
+        }
+    }
+
+    //// batch sells
+
+    function test_multiplexBatchSellTokenForToken_1() public explain("reverts if minBuyAmount is not satisfied") {
+        LibNativeOrder.RfqOrder memory rfqOrder = makeTestRfqOrder();
+        mintTo(address(rfqOrder.takerToken), rfqOrder.taker, rfqOrder.takerAmount);
+
+        try
+            zeroExDeployed.zeroEx.multiplexBatchSellTokenForToken(
+                dai,
+                zrx,
+                makeArray(makeRfqSubcall(rfqOrder)),
+                rfqOrder.takerAmount,
+                rfqOrder.makerAmount + 1
+            )
+        {
+            fail("did not revert");
+        } catch Error(string memory reason) {
+            assertEq(reason, "MultiplexFeature::_multiplexBatchSell/UNDERBOUGHT", "wrong revert reason");
+        } catch {
+            fail("low-level revert");
+        }
```

Comment:
Use [`vm.expectRevert`](https://book.getfoundry.sh/cheatcodes/expect-revert). In general, `try .. catch` in Solidity is an antipattern because it has many non-obvious failure modes that can result in security vulnerabilities. Of course, this is test code, but it's bad to build the habit.